🐛 firebase_core_desktop FirebaseOptions.fromMap no longer available

## Bug report

Faild to build app beacause FirebaseOptions.fromMap no longer availble in firebase_core_platform_interface, changed to ['fromPigeon'](https://github.com/firebase/flutterfire/blob/c70e66a546cf9236e728796c5b59a3d4e39caeb2/packages/firebase_core/firebase_core_platform_interface/lib/src/firebase_options.dart#L60)

```
: Error: Member not found: 'FirebaseOptions.fromMap'.
../…/lib/firebase_core_desktop.dart:23
      FirebaseOptions.fromMap(app.options.asMap),
                      ^^^^^^^
```

### Steps to reproduce

Steps to reproduce the behavior:


1. Create a new project and add 4 packages.
firebase_core: ^1.14.1
firebase_core_desktop: ^1.0.0

### Expected behavior

Should build project


### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel stable, 3.0.4, on macOS 12.3.1 21E258 darwin-arm, locale es-419)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.1)
[✓] VS Code (version 1.68.1)
[✓] Connected device (2 available)
[✓] HTTP Host Availability

• No issues found!
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.17.5
Flutter SDK 3.0.4
vendeme_client_flutter 3.0.0+1

dependencies:
- cloud_firestore 3.2.1 [cloud_firestore_platform_interface cloud_firestore_web collection firebase_core firebase_core_platform_interface flutter meta]
- cupertino_icons 1.0.4
- extended_image 6.0.2+1 [extended_image_library flutter meta]
- firebase_app_check 0.0.6+9 [firebase_app_check_platform_interface firebase_app_check_web firebase_core firebase_core_platform_interface flutter]
- firebase_auth 3.4.1 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_auth_desktop 1.0.0 [desktop_webview_auth firebase_auth firebase_auth_dart firebase_auth_platform_interface firebase_core firebase_core_dart flutter meta]
- firebase_core 1.14.1 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_core_desktop 1.0.0 [firebase_core_dart firebase_core_platform_interface flutter meta]
- fl_chart 0.55.0 [flutter equatable]
- fluro 2.0.3 [flutter]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_dotenv 5.0.2 [flutter]
- flutter_i18n 0.32.2 [flutter flutter_localizations flutter_web_plugins intl yaml xml2json path toml logging http]
- flutterfire_ui 0.4.2+2 [cloud_firestore collection crypto desktop_webview_auth email_validator firebase_auth firebase_core firebase_database firebase_dynamic_links flutter flutter_facebook_auth flutter_localizations flutter_svg google_sign_in sign_in_with_apple twitter_login]
- google_fonts 3.0.0 [flutter http path_provider crypto]
- google_maps_services_dart 1.17.1 [dio built_value built_collection]
- hive 2.2.1 [meta crypto]
- hive_flutter 1.1.0 [flutter hive path_provider path]
- package_info_plus 1.4.2 [flutter package_info_plus_platform_interface package_info_plus_linux package_info_plus_macos package_info_plus_windows package_info_plus_web]
- provider 6.0.2 [collection flutter nested]
- reactive_direct_select 0.2.0 [flutter reactive_forms direct_select]
- reactive_forms 13.0.0 [flutter intl]
- typesense 0.2.0 [http crypto equatable]
- url_strategy 0.2.0 [flutter_web_plugins]

dev dependencies:
- flutter_lints 2.0.1 [lints]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph]

transitive dependencies:
- _discoveryapis_commons 1.0.3 [http http_parser meta]
- async 2.8.2 [collection meta]
- boolean_selector 2.1.0 [source_span string_scanner]
- built_collection 5.1.1
- built_value 8.3.0 [built_collection collection fixnum meta]
- characters 1.2.0
- charcode 1.3.1
- clock 1.1.0
- cloud_firestore_platform_interface 5.5.10 [collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 2.6.19 [cloud_firestore_platform_interface collection firebase_core firebase_core_web flutter flutter_web_plugins js]
- collection 1.16.0
- crypto 3.0.1 [collection typed_data]
- desktop_webview_auth 0.0.9 [crypto flutter http]
- dio 4.0.6 [http_parser path]
- direct_select 2.0.0 [flutter]
- email_validator 2.0.1
- equatable 2.0.3 [collection meta]
- extended_image_library 3.1.4 [crypto flutter http_client_helper path path_provider]
- fake_async 1.3.0 [clock collection]
- ffi 1.1.2
- file 6.1.2 [meta path]
- firebase_app_check_platform_interface 0.0.4+3 [firebase_core flutter meta plugin_platform_interface]
- firebase_app_check_web 0.0.5+9 [firebase_app_check_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- firebase_auth_dart 1.0.0 [firebase_core_dart firebaseapis googleapis_auth http meta storagebox]
- firebase_auth_platform_interface 6.3.1 [firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 3.3.19 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core_dart 1.0.0 [collection meta]
- firebase_core_platform_interface 4.4.3 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 1.6.6 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_database 9.0.18 [firebase_core firebase_core_platform_interface firebase_database_platform_interface firebase_database_web flutter]
- firebase_database_platform_interface 0.2.1+10 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_database_web 0.2.0+17 [firebase_core firebase_core_web firebase_database_platform_interface flutter flutter_web_plugins js]
- firebase_dynamic_links 4.3.1 [firebase_core firebase_core_platform_interface firebase_dynamic_links_platform_interface flutter meta plugin_platform_interface]
- firebase_dynamic_links_platform_interface 0.2.3+6 [firebase_core flutter meta plugin_platform_interface]
- firebaseapis 0.1.2 [_discoveryapis_commons http]
- fixnum 1.0.1
- flutter_facebook_auth 4.3.0 [flutter flutter_facebook_auth_platform_interface flutter_facebook_auth_web]
- flutter_facebook_auth_platform_interface 3.1.0 [flutter plugin_platform_interface]
- flutter_facebook_auth_web 3.1.0 [flutter flutter_web_plugins js flutter_facebook_auth_platform_interface]
- flutter_localizations 0.0.0 [flutter intl characters clock collection material_color_utilities meta path vector_math]
- flutter_svg 1.0.3 [flutter meta path_drawing vector_math xml]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- google_sign_in 5.3.0 [flutter google_sign_in_android google_sign_in_ios google_sign_in_platform_interface google_sign_in_web]
- google_sign_in_android 5.2.5 [flutter google_sign_in_platform_interface]
- google_sign_in_ios 5.2.5 [flutter google_sign_in_platform_interface]
- google_sign_in_platform_interface 2.1.2 [flutter quiver]
- google_sign_in_web 0.10.1 [flutter flutter_web_plugins google_sign_in_platform_interface js]
- googleapis_auth 1.3.1 [crypto http http_parser]
- http 0.13.4 [async http_parser meta path]
- http_client_helper 2.0.2 [http]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- intl 0.17.0 [clock path]
- js 0.6.4
- lints 2.0.0
- logging 1.0.2
- matcher 0.12.11 [stack_trace]
- material_color_utilities 0.1.4
- meta 1.7.0
- nested 1.0.0 [flutter]
- package_info_plus_linux 1.0.5 [package_info_plus_platform_interface flutter path]
- package_info_plus_macos 1.3.0 [flutter]
- package_info_plus_platform_interface 1.0.2 [flutter meta plugin_platform_interface]
- package_info_plus_web 1.0.5 [flutter flutter_web_plugins http meta package_info_plus_platform_interface]
- package_info_plus_windows 1.0.5 [package_info_plus_platform_interface ffi flutter win32]
- path 1.8.1
- path_drawing 1.0.0 [vector_math meta path_parsing flutter]
- path_parsing 1.0.0 [vector_math meta]
- path_provider 2.0.9 [flutter path_provider_android path_provider_ios path_provider_linux path_provider_macos path_provider_platform_interface path_provider_windows]
- path_provider_android 2.0.12 [flutter path_provider_platform_interface]
- path_provider_ios 2.0.8 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.5 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_macos 2.0.5 [flutter path_provider_platform_interface]
- path_provider_platform_interface 2.0.3 [flutter platform plugin_platform_interface]
- path_provider_windows 2.0.5 [ffi flutter path path_provider_platform_interface win32]
- petitparser 4.4.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.2 [meta]
- process 4.2.4 [file path platform]
- quiver 3.0.1+1 [matcher]
- sign_in_with_apple 3.3.0 [flutter meta sign_in_with_apple_platform_interface sign_in_with_apple_web]
- sign_in_with_apple_platform_interface 1.0.0 [flutter plugin_platform_interface meta]
- sign_in_with_apple_web 1.0.1 [flutter flutter_web_plugins sign_in_with_apple_platform_interface js]
- sky_engine 0.0.99
- source_span 1.8.2 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- storagebox 0.1.0+3 [collection meta path xdg_directories]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- term_glyph 1.2.0
- test_api 0.4.9 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- toml 0.12.0 [meta petitparser quiver]
- twitter_login 4.2.2 [flutter crypto http]
- typed_data 1.3.0 [collection]
- vector_math 2.1.2
- win32 2.5.1 [ffi]
- xdg_directories 0.2.0+1 [meta path process]
- xml 5.3.1 [collection meta petitparser]
- xml2json 5.3.2 [xml]
```

</details>

---

/E:/fluttersdk/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_core_desktop-1.0.0/lib/firebase_core_desktop.dart(23,23): error G75B77105: Member not found: 'FirebaseOptions.fromMap'. [D:\Flutter projects\Dop_Software_Code_New_Old\DESKTOP\soft_desk\build\windows\flutter\flutter_assemble.vcxproj]
/E:/fluttersdk/flutter/.pub-cache/hosted/pub.dartlang.org/fl_toast-3.1.0/lib/src/toast.dart(199,22): warning G1E0FE241: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null. [D:\Flutter projects\Dop_Software_Code_New_Old\DESKTOP\soft_desk\build\windows\flutter\flutter_assemble.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(245,5): error MSB8066: Custom build for 'D:\Flutter projects\Dop_Software_Code_New_Old\DESKTOP\soft_desk\build\windows\CMakeFiles\a93e812d179b787517db322b48a3fb35\flutter_windows.dll.rule;D:\Flutter projects\Dop_Software_Code_New_Old\DESKTOP\soft_desk\build\windows\CMakeFiles\a2c4b94eb25b0947e81b0b0bca425a92\flutter_assemble.rule' exited with code 1. [D:\Flutter projects\Dop_Software_Code_New_Old\DESKTOP\soft_desk\build\windows\flutter\flutter_assemble.vcxproj]
Building Windows application...

--- a/packages/firebase_auth/firebase_auth_desktop/example/pubspec.yaml
+++ b/packages/firebase_auth/firebase_auth_desktop/example/pubspec.yaml
@@ -22,10 +22,10 @@ environment:
 
 dependencies:
   crypto: ^3.0.1
-  desktop_webview_auth: ^0.0.8
-  firebase_auth: ^3.3.7
+  desktop_webview_auth: ^0.0.9
+  firebase_auth: ^3.4.1
   firebase_auth_desktop: ^1.0.0
-  firebase_core: ^1.12.0
+  firebase_core: ^1.19.1
   firebase_core_desktop: ^1.0.0
   flutter:
     sdk: flutter

--- a/packages/firebase_auth/firebase_auth_desktop/pubspec.yaml
+++ b/packages/firebase_auth/firebase_auth_desktop/pubspec.yaml
@@ -10,10 +10,10 @@ environment:
 
 dependencies:
   desktop_webview_auth: ^0.0.8
-  firebase_auth: ^3.3.7
+  firebase_auth: ^3.4.1
   firebase_auth_dart: ^1.0.0
-  firebase_auth_platform_interface: ^6.1.4
-  firebase_core: ^1.12.0
+  firebase_auth_platform_interface: ^6.3.1
+  firebase_core: ^1.19.1
   firebase_core_dart: ^1.0.0
   flutter:
     sdk: flutter
@@ -21,11 +21,11 @@ dependencies:
 
 dev_dependencies:
   async: ^2.8.1
-  firebase_core_platform_interface: ^4.2.0
+  firebase_core_platform_interface: ^4.4.3
   flutter_test:
     sdk: flutter
   mockito: ^5.0.0
-  plugin_platform_interface: ^2.0.2
+  plugin_platform_interface: ^2.1.2
 
 flutter:
   plugin:

--- a/packages/firebase_auth/firebase_auth_desktop/test/mock.dart
+++ b/packages/firebase_auth/firebase_auth_desktop/test/mock.dart
@@ -13,36 +13,7 @@ typedef Callback = void Function(MethodCall call);
 void setupFirebaseAuthMocks([Callback? customHandlers]) {
   TestWidgetsFlutterBinding.ensureInitialized();
 
-  MethodChannelFirebase.channel.setMockMethodCallHandler((call) async {
-    if (call.method == 'Firebase#initializeCore') {
-      return [
-        {
-          'name': defaultFirebaseAppName,
-          'options': {
-            'apiKey': '123',
-            'appId': '123',
-            'messagingSenderId': '123',
-            'projectId': '123',
-          },
-          'pluginConstants': {},
-        }
-      ];
-    }
-
-    if (call.method == 'Firebase#initializeApp') {
-      return {
-        'name': call.arguments['appName'],
-        'options': call.arguments['options'],
-        'pluginConstants': {},
-      };
-    }
-
-    if (customHandlers != null) {
-      customHandlers(call);
-    }
-
-    return null;
-  });
+  setupFirebaseCoreMocks();
 }
 
 Future<T> neverEndingFuture<T>() async {

--- a/packages/firebase_core/firebase_core_dart/pubspec.yaml
+++ b/packages/firebase_core/firebase_core_dart/pubspec.yaml
@@ -5,7 +5,7 @@ repository: https://github.com/invertase/flutterfire_desktop
 version: 1.0.0
 
 environment:
-  sdk: '>=2.12.0 <3.0.0'
+  sdk: '>=2.17.0 <3.0.0'
 
 dependencies:
   collection: ^1.15.0

--- a/packages/firebase_core/firebase_core_desktop/example/macos/Runner.xcodeproj/project.pbxproj
+++ b/packages/firebase_core/firebase_core_desktop/example/macos/Runner.xcodeproj/project.pbxproj
@@ -184,6 +184,7 @@
 				33CC10EB2044A3C60003C045 /* Resources */,
 				33CC110E2044A8840003C045 /* Bundle Framework */,
 				3399D490228B24CF009A79C7 /* ShellScript */,
+				DA520B312D63141E12E60EAF /* [CP] Embed Pods Frameworks */,
 			);
 			buildRules = (
 			);
@@ -310,6 +311,23 @@
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
 			shellScript = "diff \"${PODS_PODFILE_DIR_PATH}/Podfile.lock\" \"${PODS_ROOT}/Manifest.lock\" > /dev/null\nif [ $? != 0 ] ; then\n    # print error to STDERR\n    echo \"error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.\" >&2\n    exit 1\nfi\n# This output is used by Xcode 'outputs' to avoid re-running this script phase.\necho \"SUCCESS\" > \"${SCRIPT_OUTPUT_FILE_0}\"\n";
+			showEnvVarsInLog = 0;
+		};
+		DA520B312D63141E12E60EAF /* [CP] Embed Pods Frameworks */ = {
+			isa = PBXShellScriptBuildPhase;
+			buildActionMask = 2147483647;
+			files = (
+			);
+			inputFileListPaths = (
+				"${PODS_ROOT}/Target Support Files/Pods-Runner/Pods-Runner-frameworks-${CONFIGURATION}-input-files.xcfilelist",
+			);
+			name = "[CP] Embed Pods Frameworks";
+			outputFileListPaths = (
+				"${PODS_ROOT}/Target Support Files/Pods-Runner/Pods-Runner-frameworks-${CONFIGURATION}-output-files.xcfilelist",
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "\"${PODS_ROOT}/Target Support Files/Pods-Runner/Pods-Runner-frameworks.sh\"\n";
 			showEnvVarsInLog = 0;
 		};
 /* End PBXShellScriptBuildPhase section */

--- a/packages/firebase_core/firebase_core_desktop/lib/firebase_core_desktop.dart
+++ b/packages/firebase_core/firebase_core_desktop/lib/firebase_core_desktop.dart
@@ -18,9 +18,23 @@ class FirebaseCore extends FirebasePlatform {
   }
 
   FirebaseApp _mapDartToPlatfromApp(core_dart.FirebaseApp app) {
+    final options = app.options;
+
     return FirebaseApp._(
       app.name,
-      FirebaseOptions.fromMap(app.options.asMap),
+      FirebaseOptions(
+        apiKey: options.apiKey,
+        appId: options.appId,
+        messagingSenderId: options.messagingSenderId,
+        authDomain: options.authDomain,
+        projectId: options.projectId,
+        databaseURL: options.databaseURL,
+        measurementId: options.measurementId,
+        storageBucket: options.storageBucket,
+        trackingId: options.trackingId,
+        appGroupId: options.appGroupId,
+        deepLinkURLScheme: options.deepLinkURLScheme,
+      ),
     );
   }
 

--- a/packages/firebase_core/firebase_core_desktop/pubspec.yaml
+++ b/packages/firebase_core/firebase_core_desktop/pubspec.yaml
@@ -5,31 +5,31 @@ repository: https://github.com/invertase/flutterfire_desktop
 version: 1.0.0
 
 environment:
-  sdk: '>=2.12.0 <3.0.0'
+  sdk: '>=2.17.0 <3.0.0'
   flutter: '>=1.20.0'
 
 dependencies:
   firebase_core_dart: ^1.0.0
-  firebase_core_platform_interface: ^4.1.0
+  firebase_core_platform_interface: ^4.4.3
   flutter:
     sdk: flutter
   meta: ^1.3.0
 
 dev_dependencies:
-  firebase_core: ^1.10.0
+  firebase_core: ^1.19.1
   flutter_test:
     sdk: flutter
   mockito: ^5.0.0
-  plugin_platform_interface: ^2.0.2
+  plugin_platform_interface: ^2.1.2
 
 flutter:
   plugin:
     implements: firebase_core
     platforms:
       # TODO when doing development on macOS, uncomment this.
-      # macos:
-      #   dartPluginClass: FirebaseCore
-      #   pluginClass: none
+      macos:
+        dartPluginClass: FirebaseCore
+        pluginClass: none
       linux:
         dartPluginClass: FirebaseCore
         pluginClass: none

--- a/packages/firebase_functions/firebase_functions_desktop/example/linux/flutter/generated_plugins.cmake
+++ b/packages/firebase_functions/firebase_functions_desktop/example/linux/flutter/generated_plugins.cmake
@@ -5,6 +5,9 @@
 list(APPEND FLUTTER_PLUGIN_LIST
 )
 
+list(APPEND FLUTTER_FFI_PLUGIN_LIST
+)
+
 set(PLUGIN_BUNDLED_LIBRARIES)
 
 foreach(plugin ${FLUTTER_PLUGIN_LIST})
@@ -13,3 +16,8 @@ foreach(plugin ${FLUTTER_PLUGIN_LIST})
   list(APPEND PLUGIN_BUNDLED_LIBRARIES $<TARGET_FILE:${plugin}_plugin>)
   list(APPEND PLUGIN_BUNDLED_LIBRARIES ${${plugin}_bundled_libraries})
 endforeach(plugin)
+
+foreach(ffi_plugin ${FLUTTER_FFI_PLUGIN_LIST})
+  add_subdirectory(flutter/ephemeral/.plugin_symlinks/${ffi_plugin}/linux plugins/${ffi_plugin})
+  list(APPEND PLUGIN_BUNDLED_LIBRARIES ${${ffi_plugin}_bundled_libraries})
+endforeach(ffi_plugin)

--- a/packages/firebase_functions/firebase_functions_desktop/example/windows/flutter/generated_plugins.cmake
+++ b/packages/firebase_functions/firebase_functions_desktop/example/windows/flutter/generated_plugins.cmake
@@ -5,6 +5,9 @@
 list(APPEND FLUTTER_PLUGIN_LIST
 )
 
+list(APPEND FLUTTER_FFI_PLUGIN_LIST
+)
+
 set(PLUGIN_BUNDLED_LIBRARIES)
 
 foreach(plugin ${FLUTTER_PLUGIN_LIST})
@@ -13,3 +16,8 @@ foreach(plugin ${FLUTTER_PLUGIN_LIST})
   list(APPEND PLUGIN_BUNDLED_LIBRARIES $<TARGET_FILE:${plugin}_plugin>)
   list(APPEND PLUGIN_BUNDLED_LIBRARIES ${${plugin}_bundled_libraries})
 endforeach(plugin)
+
+foreach(ffi_plugin ${FLUTTER_FFI_PLUGIN_LIST})
+  add_subdirectory(flutter/ephemeral/.plugin_symlinks/${ffi_plugin}/windows plugins/${ffi_plugin})
+  list(APPEND PLUGIN_BUNDLED_LIBRARIES ${${ffi_plugin}_bundled_libraries})
+endforeach(ffi_plugin)

--- a/packages/firebase_functions/firebase_functions_desktop/pubspec.yaml
+++ b/packages/firebase_functions/firebase_functions_desktop/pubspec.yaml
@@ -9,9 +9,9 @@ environment:
   flutter: '>=1.20.0'
 
 dependencies:
-  cloud_functions: ^3.1.1
-  cloud_functions_platform_interface: ^5.0.14
-  firebase_core: ^1.10.0
+  cloud_functions: ^3.3.1
+  cloud_functions_platform_interface: ^5.1.10
+  firebase_core: ^1.19.1
   firebase_core_dart: ^1.0.0
   firebase_core_desktop: ^1.0.0
   firebase_functions_dart: ^0.1.1+1
@@ -21,20 +21,20 @@ dependencies:
   meta: ^1.3.0
 
 dev_dependencies:
-  firebase_core_platform_interface: ^4.1.0
+  firebase_core_platform_interface: ^4.4.3
   flutter_test:
     sdk: flutter
   mockito: ^5.0.0
-  plugin_platform_interface: ^2.0.2
+  plugin_platform_interface: ^2.1.2
 
 flutter:
   plugin:
     implements: cloud_functions
     platforms:
       # TODO when doing development on macOS, uncomment this.
-      # macos:
-      #   dartPluginClass: FirebaseFunctionsDesktop
-      #   pluginClass: none
+      macos:
+        dartPluginClass: FirebaseFunctionsDesktop
+        pluginClass: none
       linux:
         dartPluginClass: FirebaseFunctionsDesktop
         pluginClass: none

